### DISCUSSION TITLE
How do you get these static values of the variable `rgbVar` in the file `Augmentation.py`?

### DISCUSSION THREAD

**Discussion by fzh0917** [STVIR/pysot#312] [General]
Hi, I am curious about values of the variable `rgbVar` in the file `Augmentation.py`, they are here:
```Python
self.rgbVar = np.array(
            [[-0.55919361,  0.98062831, - 0.41940627],
             [1.72091413,  0.19879334, - 1.82968581],
             [4.64467907,  4.73710203, 4.88324118]], dtype=np.float32)
```

Maybe the principle is in [here](https://github.com/EvilDJ/target_tracking/blob/ce013c67771a013856d80071cd65809e522909a7/%E8%BF%BD%E8%B8%AA_simfc/tensorflow-siamese-fc-master/train_simple.py#L39)，but I don't know which data you used to generate these static values for speeding up data loading.

Would you please describe it in detail? Thanks!

**Comment by fzh0917** [comment]
# The second problem
In the file `augmentation.py`, the code of color augmentation is that:
```Python
    def _color_aug(self, image):
        offset = np.dot(self.rgbVar, np.random.randn(3, 1))
        offset = offset[::-1]  # bgr 2 rgb
        offset = offset.reshape(3)
        image = image - offset
        return image
```
The channel order is `BGR` if using `cv2.imread` to read images, but the comment of the line `offset = offset[::-1]` says "bgr 2 rgb", is this wrong? 

More precisely, if the argument `gray`  of the function `__call__` is set to `True`, then the function `_gray_aug` will execute, and the above problem will not matter, because the channel order `BGR` is same as `RGB`. However, if setting the argument `gray` to `False`, then the channel order `BGR` is  different to `RGB`, so the above problem will appear.

**Comment by ZhiyuanChen** [comment]
> ```python
> [::-1]
> ```

https://www.scivision.co/numpy-image-bgr-to-rgb/

**Comment by fzh0917** [comment]
Thank you!
Well, this is not what I want to ask in the second problem.^_^ 
My initial question is about order of color channels for data augmentation. However, I have figured the above problems out now. The order of color channels described in the first problem is completely correct. :)
Thank you and all of your doing.
Best regards!
